ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PullQueryPublisher.java

```diff
@@ -79,98 +78,68 @@
 
   @Override
   public synchronized void subscribe(final Subscriber<Collection<StreamedRow>> subscriber) {
-    final PullQuerySubscription subscription = new PullQuerySubscription(
-        subscriber,
-        () -> {
-          final RoutingOptions routingOptions = new PullQueryConfigRoutingOptions(
-              query.getSessionConfig().getConfig(false),
-              query.getSessionConfig().getOverrides(),
-              ImmutableMap.of()
-          );
-
-          PullQueryExecutionUtil.checkRateLimit(rateLimiter);
-
-          final PullQueryResult result = ksqlEngine.executePullQuery(
-              serviceContext,
-              query,
-              routing,
-              routingFilterFactory,
-              routingOptions,
-              pullQueryMetrics
-          );
-
-          pullQueryMetrics.ifPresent(pullQueryExecutorMetrics -> pullQueryExecutorMetrics
-              .recordLatency(startTimeNanos));
-          return result;
-        },
-        query
+    final RoutingOptions routingOptions = new PullQueryConfigRoutingOptions(
+        query.getSessionConfig().getConfig(false),
+        query.getSessionConfig().getOverrides(),
+        ImmutableMap.of()
+    );
+
+    PullQueryExecutionUtil.checkRateLimit(rateLimiter);
+
+    final PullQueryResult result = ksqlEngine.executePullQuery(
```

Comment:
previously we were calling this inside the `Subscription#request` method, now we're doing it before calling the subscriber. I guess that this is intentional because of the new asynchronous nature of pull queries? Just wanted to confirm that I understand correctly (and to confirm that it is OK to do this here in the `subscribe` method instead of on the first `poll` call)